src/lib.rs

```diff
@@ -286,6 +287,7 @@ pub fn get_package_response(
     match parts.status {
         StatusCode::OK => (),
         StatusCode::FORBIDDEN => return Err(ApiError::NotFound), // Oddly this is the not-found code
+        StatusCode::NOT_FOUND => return Err(ApiError::NotFound),
```

Comment:
This line is actually unrelated to the dependency upgrade. It is changed because now the `get_package_response` and `get_package_tarball_response` functions return `UnexpectedResponse` instead of `NotFound`, I modified these two functions. Should I split the current PR into two?

Reply:
Yes please, another PR. We'll need tests for it too